pkg/scheduler/scheduler.go

```diff
@@ -469,42 +510,62 @@ func updatePoolAllocation(ext extensions.ExtensionsClientset, pool *extensions.I
 
 	glog.V(2).Infof("Update pool with object %v", pool)
 	_, err := ext.IPClaimPools().Update(pool)
-	if err != nil {
-		return err
-	}
-	return nil
+	return err
 }
 
 func addServiceExternalIP(svc *v1.Service, kcs kubernetes.Interface, ip string) error {
 	glog.V(5).Infof(
-		"Try to update service's %v external IPs list with address %v",
+		"Try to update service's %v externalIPs list with address %v",
 		svc.ObjectMeta.Name, ip,
 	)
 	svc.Spec.ExternalIPs = append(svc.Spec.ExternalIPs, ip)
 	_, err := kcs.Core().Services(svc.ObjectMeta.Namespace).Update(svc)
-	if err != nil {
-		return err
+	return err
+}
+
+func tryUpdateIPClaim(ext extensions.ExtensionsClientset, claim *extensions.IpClaim) error {
+	retries := 3
+	for {
+		_, err := ext.IPClaims().Update(claim)
+		if err == nil {
+			return nil
+		}
+		if !apierrors.IsConflict(err) || retries -= 1; retries == 0 {
+			return err
+		}
 	}
-	return nil
 }
 
-func tryCreateIPClaim(ext extensions.ExtensionsClientset, ipclaim *extensions.IpClaim) error {
-	_, err := ext.IPClaims().Create(ipclaim)
+func tryCreateIPClaim(ext extensions.ExtensionsClientset, new_claim *extensions.IpClaim) error {
+	_, err := ext.IPClaims().Create(new_claim)
 	if apierrors.IsAlreadyExists(err) {
-		return nil
+		// add new owner to claim's OwnerReferences
+		claim, err := ext.IPClaims().Get(new_claim.Metadata.Name)
+		if err != nil {
+			return err
+		}
+		claim.Metadata.OwnerReferences = append(claim.Metadata.OwnerReferences, new_claim.Metadata.OwnerReferences[0])
+		err = tryUpdateIPClaim(ext, claim)
```

Comment:
we may fail to update claim, because of concurrent delete request. In such case we need to create new ipclaim with single owner